internal/pkg/mapper/mapper.go

```diff
@@ -0,0 +1,30 @@
+package mapper
+
+import (
+	"fmt"
+
+	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	aggregationv1 "github.com/envoyproxy/xds-relay/pkg/api/aggregation/v1"
+)
+
+// Mapper defines the interface that Maps an incoming request to an aggregation key
+type Mapper interface {
+	// GetKey converts a request into an aggregated key
+	GetKey(request v2.DiscoveryRequest, typeURL string) (string, error)
```

Comment:
I have some concerns about including typeURL here because there is a typeURL in the discovery request. How do we handle ambiguity when it's supplied in both places? ADS will still wrap the discovery request no? https://github.com/envoyproxy/envoy/blob/master/api/envoy/service/discovery/v2/ads.proto#L27

Reply:
https://github.com/envoyproxy/go-control-plane/blob/master/pkg/server/server.go#L53
TypeUrl will be "" for ADS. and then we have these kind of sspecial casing for ADS https://github.com/envoyproxy/go-control-plane/blob/master/pkg/server/server.go#L305-L311

https://github.com/envoyproxy/envoy/blob/master/api/envoy/api/v2/discovery.proto#L43-L46

All of these suggest that typeurl could be empty. We probably do not want the concept of typeUrl to leak into other components. Im imaging the typeUrl to never be empty in `Mapper.GetKey`

Reply:
Also notice ADS is not a kind of resource https://github.com/envoyproxy/go-control-plane/blob/master/pkg/cache/resource.go#L36-L43

Reply:
Added more comments https://github.com/envoyproxy/xds-relay/pull/19/files#diff-6d8632086beef4726597650f47c0bec7R15-R17